### PR TITLE
8344795: Remove uses of AccessControlContext in java.desktop module

### PR DESCRIPTION
**Pull request by prrace**
This removes uses of AccessControlContext in java.desktop

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8344795](https://bugs.openjdk.org/browse/JDK-8344795): Remove uses of AccessControlContext in java.desktop module (**Bug** - P4)

### Reviewers
 * [Alexander Zvegintsev](https://openjdk.org/census#azvegint) (@azvegint - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22308/head:pull/22308` \
`$ git checkout pull/22308`

Update a local copy of the PR: \
`$ git checkout pull/22308` \
`$ git pull https://git.openjdk.org/jdk.git pull/22308/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22308`

View PR using the GUI difftool: \
`$ git pr show -t 22308`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22308.diff">https://git.openjdk.org/jdk/pull/22308.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22308#issuecomment-2492376107)
</details>

### CODE REVIEW

File: src/java.desktop/share/classes/java/awt/Component.java [hunk 1]
```diff
@@ -8974,8 +8949,6 @@ private void readObject(ObjectInputStream s)
     {
```
Comment (azvegint) [inline src/java.desktop/share/classes/java/awt/Component.java:8949]: `@SuppressWarnings("removal")` can be removed

File: src/java.desktop/share/classes/java/awt/MenuComponent.java [hunk 2]
```diff
@@ -449,8 +428,6 @@ private void readObject(ObjectInputStream s)
     {
```
Comment (azvegint) [inline src/java.desktop/share/classes/java/awt/MenuComponent.java:428]: `@SuppressWarnings("removal")` can be removed

File: src/java.desktop/share/classes/javax/swing/UIDefaults.java [hunk 3]
```diff
@@ -1119,7 +1115,6 @@ public ProxyLazyValue(String c, Object[] o) {
          */
         @SuppressWarnings("removal")
```
Comment (azvegint) [inline src/java.desktop/share/classes/javax/swing/UIDefaults.java:1116]:
```suggestion
```

**Review by azvegint — COMMENTED** [review]
I think that the removal of the `AccessControlException` also belongs to this PR

there is only one usage:
```
./src/java.desktop/share/classes/java/awt/Dialog.java:36:import java.security.AccessControlException;
./src/java.desktop/share/classes/java/awt/Dialog.java:1597:        } catch (@SuppressWarnings("removal") AccessControlException ace) {
``` 

And it seems that the `java.awt.Dialog#checkModalityPermission()` has became obsolete.

File: src/java.desktop/share/classes/java/awt/Component.java (hunk 1 above)
Comment (prrace) [reply src/java.desktop/share/classes/java/awt/Component.java:8949]: ok

File: src/java.desktop/share/classes/java/awt/MenuComponent.java (hunk 2 above)
Comment (prrace) [reply src/java.desktop/share/classes/java/awt/MenuComponent.java:428]: ok

File: src/java.desktop/share/classes/javax/swing/UIDefaults.java (hunk 3 above)
Comment (prrace) [reply src/java.desktop/share/classes/javax/swing/UIDefaults.java:1116]: ok

**Comment by prrace** [comment]
> I think the removal of the use of the `AccessControlException` should also be part of this PR.
> 
> there is only one usage:
> 
> ```
> ./src/java.desktop/share/classes/java/awt/Dialog.java:36:import java.security.AccessControlException;
> ./src/java.desktop/share/classes/java/awt/Dialog.java:1597:        } catch (@SuppressWarnings("removal") AccessControlException ace) {
> ```
> 
> And it seems that the `java.awt.Dialog#checkModalityPermission()` has became obsolete.

I deliberately left these alone, because it is part of removing most AWTPermissions which I have partially prepared already for a separate PR.
The fix will come soon, perhaps tomorrow,  perhaps Monday.

**Review by azvegint — APPROVED** [review]

**Pull request closed without merging**